File: web/src/api/lg/order.js
import service from '@/utils/request'
import { ElMessage } from 'element-plus'

export const getOrderList = (params) => {
  return service({
    url: '/order/getOrderList',
    method: 'get',
    params
  })
}

export const approveApply = (data) => {
  return service({
    url: '/order/approveApply',
    method: 'put',
    data
  })
}

export const rejectApply = (data) => {
  return service({
    url: '/order/rejectApply',
    method: 'put',
    data
  })
}

export const approveDelay = (data) => {
  return service({
    url: '/order/approveDelay',
    method: 'put',
    data
  })
}

export const rejectDelay = (data) => {
  return service({
    url: '/order/rejectDelay',
    method: 'put',
    data
  })
}

export const approveClaim = (data) => {
  return service({
    url: '/order/approveClaim',
    method: 'put',
    data
  })
}

export const rejectClaim = (data) => {
  return service({
    url: '/order/rejectClaim',
    method: 'put',
    data
  })
}

export const openLetter = (data) => {
  return service({
    url: '/order/openLetter',
    method: 'put',
    data
  })
}

export const rePush = (data) => {
  return service({
    url: '/order/rePush',
    method: 'put',
    data
  })
}

export const getInsuranceBalance = (params) => {
  return service({
    url: '/order/getInsuranceBalance',
    method: 'get',
    params
  })
}

export const getOrderStatisticData = (params) => {
  return service({
    url: '/order/getOrderStatisticData',
    method: 'get',
    params
  })
}

export const getEmployeeStatisticData = (params) => {
  return service({
    url: '/order/getEmployeeStatisticData',
    method: 'get',
    params
  })
}

export const getGEODistributionData = (params) => {
  return service({
    url: '/order/getGEODistributionData',
    method: 'get',
    params
  })
}

export const getOrderTrendData = (params) => {
  return service({
    url: '/order/getOrderTrendData',
    method: 'get',
    params
  })
}

export const downloadExcel = (params) => {
  return service({
    url: '/order/exportExcel',
    method: 'get',
    params: params,
    responseType: 'blob'
  }).then((res) => {
    handleExcelError(res)
  })
}

const handleExcelError = (res) => {
  if (typeof (res.data) !== 'undefined') {
    if (res.data.type === 'application/json') {
      const reader = new FileReader()
      reader.onload = function() {
        const message = JSON.parse(reader.result).msg
        ElMessage({
          showClose: true,
          message: message,
          type: 'error'
        })
      }
      reader.readAsText(new Blob([res.data]))
    }
  } else {
    const downloadUrl = window.URL.createObjectURL(new Blob([res]))
    const a = document.createElement('a')
    a.style.display = 'none'
    a.href = downloadUrl
    a.download = (new Date().getTime()).toString() + '.xlsx'
    a.click()
  }
}

export const assignOrder = (data) => {
  return service({
    url: '/order/assignOrder',
    method: 'put',
    data
  })
}

export const markOfflineRefund = (data) => {
  return service({
    url: '/order/markOfflineRefund',
    method: 'post',
    data
  })
}

export const unmarkOfflineRefund = (data) => {
  return service({
    url: '/order/unmarkOfflineRefund',
    method: 'post',
    data
  })
}

export const downloadInvoiceExcel = (params) => {
  return service({
    url: '/order/exportInvoiceExcel',
    method: 'get',
    params: params,
    responseType: 'blob'
  }).then((res) => {
    handleExcelError(res)
  })
}

export const elogValidate = (data) => {
  return service({
    url: '/order/elogValidate',
    method: 'post',
    data
  })
}

File: server/router/lg/order.go
package lg

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type OrderRouter struct {
}

func (s *OrderRouter) InitOrderRouter(Router *gin.RouterGroup) {
	orderRouter := Router.Group("order").Use(middleware.OperationRecord())
	orderRouterWithoutRecord := Router.Group("order")
	var orderApi = v1.ApiGroupApp.LgApiGroup.OrderApi
	{
		orderRouter.PUT("approveApply", orderApi.ApproveApply) // 审批同意申请
		orderRouter.PUT("rejectApply", orderApi.RejectApply)   // 审批拒绝申请
		orderRouter.PUT("approveDelay", orderApi.ApproveDelay) // 审批同意延期
		orderRouter.PUT("rejectDelay", orderApi.RejectDelay)   // 审批拒绝延期
		orderRouter.PUT("approveClaim", orderApi.ApproveClaim) // 审批同意理赔
		orderRouter.PUT("rejectClaim", orderApi.RejectClaim)   // 审批拒绝理赔
		orderRouter.PUT("openLetter", orderApi.OpenLetter)     // 提交开函申请
		orderRouter.PUT("rePush", orderApi.RePush)             // 提交重推申请
		orderRouter.PUT("assignOrder", orderApi.AssignOrder)   // 订单认领
	}
	{
		orderRouterWithoutRecord.GET("getOrderList", orderApi.GetOrderList)                         // 获取Order列表
		orderRouterWithoutRecord.GET("getInsuranceBalance", orderApi.GetInsuranceBalance)           // 获取担保额度
		orderRouterWithoutRecord.GET("getOrderStatisticData", orderApi.GetOrderStatisticData)       // 获取Order统计数据
		orderRouterWithoutRecord.GET("getEmployeeStatisticData", orderApi.GetEmployeeStatisticData) // 获取Order开单数据
		orderRouterWithoutRecord.GET("getGEODistributionData", orderApi.GetGEODistributionData)     // 获取Order地理分布数据
		orderRouterWithoutRecord.GET("getOrderTrendData", orderApi.GetOrderTrendData)               // 获取Order趋势数据
		orderRouterWithoutRecord.GET("exportExcel", orderApi.ExportExcel)                           // 导出Order数据到excel
		orderRouterWithoutRecord.POST("markOfflineRefund", orderApi.MarkOfflineRefund)              // 绑定项目
		orderRouterWithoutRecord.POST("unmarkOfflineRefund", orderApi.UnmarkOfflineRefund)          // 解绑项目
		orderRouterWithoutRecord.GET("exportInvoiceExcel", orderApi.ExportInvoiceExcel)             // 导出Invoice数据到excel
	}
}

func (s *OrderRouter) InitPublicOrderRouter(Router *gin.RouterGroup) {
	orderRouterWithoutRecord := Router.Group("order")
	var orderApi = v1.ApiGroupApp.LgApiGroup.OrderApi
	orderRouterWithoutRecord.POST("elogValidate", orderApi.ElogValidate) // 保函鉴真
}
